src/V4/PeanutRouter.sol

```diff
@@ -0,0 +1,120 @@
+     * transfers getting stuck in Axelar (Squid works on top of Axelar) in case gas price
+     * on the destination chain increases while the tx is pending in the mempool on the source chain.
+     * But it doesn't really matter since all transfers are paid by us and we can always increase
+     * the axelar fee (e.g. from our backend server).
+     * @param _depositIndex The index of the deposit in peanut.
+     * @param _withdrawalSignature signature to withdraw from peanut.
+     * @param _squidFee squid router fee.
+     * @param _peanutFee fee amount taken by peanut (this contract) for routing.
+     * @param _squidData calldata fro the squid router
+     * @param _routingSignature signed _squidFee, _peanutFee and _squidData
+     */
+    function withdrawAndBridge(
+        uint256 _depositIndex,
+        bytes calldata _withdrawalSignature,
+        uint256 _squidFee,
+        uint256 _peanutFee,
+        bytes calldata _squidData,
+        bytes calldata _routingSignature
+    ) public payable {
+        PeanutV4.Deposit memory deposit = peanut.getDeposit(_depositIndex);
+
+        // We must first validate _routingSignature to prevent front-running
+        // The signature structure follows version 0x00 from EIP-191
+        bytes32 digest = keccak256(
+            abi.encodePacked(
+                bytes2(0x0019),
+                address(this),
+                block.chainid,
+                address(peanut),
+                _depositIndex,
+                squidAddress,
+                _squidFee,
+                _peanutFee,
+                _squidData
+            )
+        );
+        address routingSigner = ECDSA.recover(digest, _routingSignature);
+        require(routingSigner == deposit.pubKey20, "WRONG ROUTING SIGNER");
+
+        require(_squidFee == msg.value, "msg.value MUST BE THE SQUID FEE");
+        require(deposit.contractType == 0 || deposit.contractType == 1, "UNSUPPORTED contractType");
+        require(_peanutFee < deposit.amount, "TOO HIGH FEE");
+
+        peanut.withdrawDepositAsRecipient(_depositIndex, address(this), _withdrawalSignature);
+
+        uint256 amountToBridge = deposit.amount - _peanutFee;
+        uint256 ethAmountToSquid = msg.value;
+        if (deposit.contractType == 0) { // ETH deposit
+            ethAmountToSquid += amountToBridge;
+        } else if (deposit.contractType == 1) { // ERC20 deposit
+            IERC20 token = IERC20(deposit.tokenAddress);
+            token.safeApprove(address(squidAddress), amountToBridge);
+        } else {
+            revert("UNSUPPORTED contractType");
+        }
+
+        // execute the cross-chain transfer
+        (bool success,) = payable(squidAddress).call{value: ethAmountToSquid}(_squidData);
+        require(success, "SQUID DEPOSIT DID NOT SUCCEED");
+    }
+
+    function withdrawFees(address token, address to, uint256 amount) public onlyOwner {
+        if (token == address(0)) {
+            payable(to).transfer(amount);
+        } else {
```

Comment:
I'd use call instead of transfer. It could be that receiver is a smart contract wallet, and for some of them the 2300 gas budget is not enough

Reply:
Well, it's solely for us, but still makes sense 👍